spring-integration-core/src/main/java/org/springframework/integration/handler/support/MessagingMethodInvokerHelper.java

```diff
@@ -818,14 +818,19 @@ private HandlerMethod obtainHandlerMethodIfAny(Method methodToProcess) {
 	}
 
 	private boolean isMethodEligible(Method methodToProcess) {
-		return !(methodToProcess.isBridge() || // NOSONAR boolean complexity
+		return (!(methodToProcess.isBridge() || // NOSONAR boolean complexity
 				isMethodDefinedOnObjectClass(methodToProcess) ||
 				methodToProcess.getDeclaringClass().equals(Proxy.class) ||
 				(this.requiresReply && void.class.equals(methodToProcess.getReturnType())) ||
-				(this.methodName != null && !this.methodName.equals(methodToProcess.getName())) ||
-				(this.methodName == null &&
-						ObjectUtils.containsElement(new String[] { "start", "stop", "isRunning" },
-								methodToProcess.getName())));
+				(this.methodName != null && !this.methodName.equals(methodToProcess.getName()))))
+			&& !isPausableMethod(methodToProcess);
+	}
+
+	private boolean isPausableMethod(Method pausableMethod) {
+		Class<?> declaringClass = pausableMethod.getDeclaringClass();
+		return (Pausable.class.isAssignableFrom(declaringClass) || Lifecycle.class.isAssignableFrom(declaringClass))
```

Comment:
Any chances that we log special info that `Pausable` methods are not considered as candidates?